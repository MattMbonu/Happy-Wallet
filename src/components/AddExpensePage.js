import React, { Component } from "react";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/Expenses";
import { connect } from "react-redux";

export class AddExpensePage extends Component {
  onSubmit = expense => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <h1>This is my Create Expense Page page!!</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispactchToProps = dispatch => {
  return {
    addExpense: expense => dispatch(addExpense(expense))
  };
};
export default connect(
  undefined,
  mapDispactchToProps
)(AddExpensePage);
