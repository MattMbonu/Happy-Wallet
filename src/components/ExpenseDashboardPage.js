import React from "react";
import ConnectedExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpenseDashboardPage = () => (
  <div>
    <p>this is my expenses Dashboard</p>
    <ExpenseListFilters />
    <ConnectedExpenseList />
  </div>
);

export default ExpenseDashboardPage;
