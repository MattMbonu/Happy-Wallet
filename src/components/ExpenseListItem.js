import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

export const ExpenseListItem = props => {
  return (
    <div>
      <Link to={`/edit/${props.id}`}>
        <h1>{props.description}</h1>
      </Link>
      <h2>
        {numeral(props.amount / 100).format("$0,0.00")}- Date:
        {moment(props.createdAt).format("MMMM Do, YYYY")}
      </h2>
    </div>
  );
};

export default connect()(ExpenseListItem);
