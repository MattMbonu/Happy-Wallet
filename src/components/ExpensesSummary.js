import React from "react";
import numeral from "numeral";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

const ExpensesSummary = props => (
  <div>
    <h1>
      Viewing: {props.expenses.length}{" "}
      {props.expenses.length === 1 ? "expense" : "expenses"} totalling{" "}
      {numeral(props.expensesTotal / 100).format("$0,0.00")}
    </h1>
  </div>
);

const mapStateToProps = state => {
  const selectedExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenses: selectedExpenses,
    expensesTotal: selectExpensesTotal(selectedExpenses)
  };
};
export default connect(mapStateToProps)(ExpensesSummary);
