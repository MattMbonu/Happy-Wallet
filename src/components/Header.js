import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Happy Wallet</h1>
    <p>welcome to the happiest place on earth... for your bank account!</p>
    <NavLink exact={true} activeClassName="is-active" to="/">
      Dashboard
    </NavLink>{" "}
    <NavLink activeClassName="is-active" to="/create">
      Create an Expense
    </NavLink>
    <NavLink activeClassName="is-active" to="/help">
      Help
    </NavLink>
  </header>
);

export default Header;
