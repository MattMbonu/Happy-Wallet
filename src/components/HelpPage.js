import React from "react";

const HelpPage = () => (
  <div>
    <h1>This is my help page!!</h1>
    <p>this is my expenses</p>
  </div>
);

export default HelpPage;
