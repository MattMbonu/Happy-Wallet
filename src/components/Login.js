import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const Login = props => (
  <div>
    <p>Login with your google account to get started!</p>
    <button onClick={props.startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});
export default connect(
  undefined,
  mapDispatchToProps
)(Login);
