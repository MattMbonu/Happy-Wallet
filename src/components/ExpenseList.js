import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";
import { Link } from "react-router-dom";

export const ExpenseList = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>
        no expenses click{" "}
        <Link to="/create" exact={true}>
          here
        </Link>{" "}
        to create a new expense
      </p>
    ) : (
      props.expenses.map(expense => (
        <ExpenseListItem
          key={expense.id}
          description={expense.description}
          amount={expense.amount}
          createdAt={expense.createdAt}
          id={expense.id}
        />
      ))
    )}
  </div>
);

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters)
});
export default connect(mapStateToProps)(ExpenseList);
