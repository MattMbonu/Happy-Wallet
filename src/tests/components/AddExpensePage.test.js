import React from "react";
import { AddExpensePage } from "../../components/AddExpensePage";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";

test("should render page", () => {
  const addExpenseSpy = jest.fn();
  const history = { push: jest.fn() };
  const wrapper = shallow(
    <AddExpensePage addExpense={addExpenseSpy} history={history} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should handle onSubmit", () => {
  const addExpenseSpy = jest.fn();
  const history = { push: jest.fn() };
  const wrapper = shallow(
    <AddExpensePage addExpense={addExpenseSpy} history={history} />
  );
  wrapper.find("ExpenseForm").prop("onSubmit")(expenses[0]);
  expect(addExpenseSpy).toHaveBeenLastCalledWith(expenses[0]);
  expect(history.push).toHaveBeenLastCalledWith("/");
});
