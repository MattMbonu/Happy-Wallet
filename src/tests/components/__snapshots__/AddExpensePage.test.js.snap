// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page 1`] = `
<div>
  <h1>
    This is my Create Expense Page page!!
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
